tests/lib/tools/snaps-state

```diff
@@ -59,6 +62,92 @@ install_local_as() {
     install_local "$snap" --name "$name" "$@"
 }
 
+is_classic_confinement_supported() {
+    snap debug sandbox-features --required=confinement-options:classic
+}
+
+# The function will re-pack either core or snpad snap using the assets
+# from the snapd deb installed on the system
+repack_snapd_deb_into_snap() {
+    local SNAP_NAME="$1"
+
+    if  [ "$SNAP_NAME" = "core" ]; then
+        # use snapd from edge as a recent snap that should be close to what we will
+        # have in the snapd deb
+        snap download core --basename=core --edge
+        unsquashfs -d ./core-unpacked core.snap
+
+        # extract all the files from the snapd deb
+        dpkg-deb -x "$SPREAD_PATH"/../snapd_*.deb ./core-unpacked
```

Comment:
Should it take the deb as argument instead? It would not be as tightly coupled to the particular layout of the test suite this way and we call it a handful of times at most.

Reply:
yes, nice idea